src/main/java/dev/meyi/bn/utilities/EnchantedCraftingHandler.java

```diff
@@ -0,0 +1,155 @@
+                double ProfitPerMil = ProfitPerMilC * BazaarNotifier.bazaarDataRaw.getJSONObject(key1).getJSONArray("sell_summary").getJSONObject(1).getDouble("pricePerUnit") - (BazaarNotifier.bazaarDataRaw.getJSONObject(Material).getJSONArray("sell_summary").getJSONObject(1).getDouble("pricePerUnit") * 160);
+
+                list.add(new ArrayList<>(Arrays.asList(Double.toString(Price1), Double.toString(Price2), Double.toString(ProfitPerMil), key1)));
+                key1 = NormalKeys.next();
+            }catch (Exception ignored){}
+        }
+
+
+        Iterator<String> OtherKeys = BazaarNotifier.enchantCraftingList.getJSONObject("Other").keys();
+        for (int i = 0; i < BazaarNotifier.enchantCraftingList.getJSONObject("Other").length(); i++) {
+
+
+            String ItemName = OtherKeys.next();
+            double ItemSellPrice = BazaarNotifier.bazaarDataRaw.getJSONObject(ItemName).getJSONArray("sell_summary").getJSONObject(1).getDouble("pricePerUnit");
+            double ItemBuyPrice = BazaarNotifier.bazaarDataRaw.getJSONObject(ItemName).getJSONArray("buy_summary").getJSONObject(1).getDouble("pricePerUnit");
+            double IngredientP= 0d;
+            int IngredientC;
+            double P= 0d;
+
+            for(int h = 0; h < BazaarNotifier.enchantCraftingList.getJSONObject("Other").getJSONObject(ItemName).getJSONArray("Material").length(); h++){
+                if(h % 2 == 0){
+                    IngredientP = BazaarNotifier.bazaarDataRaw.getJSONObject(BazaarNotifier.enchantCraftingList.getJSONObject("Other").getJSONObject(ItemName).getJSONArray("Material").getString(h)).getJSONArray("sell_summary").getJSONObject(0).getDouble("pricePerUnit");
+                }else{
+                    IngredientC = BazaarNotifier.enchantCraftingList.getJSONObject("Other").getJSONObject(ItemName).getJSONArray("Material").getInt(h);
+                    P += (IngredientP*IngredientC);
+                }
+            }
+
+
+
+            Double Price1 = ItemSellPrice - P;
+            double Price2 = ItemBuyPrice - P;
+            Double CountPerMil = 1000000 / P;
+            double PricePerMil = CountPerMil * Price1;
+            list.add(new ArrayList<>(Arrays.asList(Price1.toString(), Double.toString(Price2), Double.toString(PricePerMil), ItemName)));
+
+        }
+
+
+
+
+        list.sort((o1, o2) -> {
+            ArrayList<Double> list1 = new ArrayList<>();
+            ArrayList<Double> list2 = new ArrayList<>();
+            list1.add(Double.valueOf(o1.get(0)));
+            list1.add(Double.valueOf(o1.get(1)));
+            list1.add(Double.valueOf(o1.get(2)));
+            list2.add(Double.valueOf(o2.get(0)));
+            list2.add(Double.valueOf(o2.get(1)));
+            list2.add(Double.valueOf(o2.get(2)));
+
+
+            return list1.get(BazaarNotifier.config.getInt("CraftingGUITOGGLE")).compareTo(list2.get(BazaarNotifier.config.getInt("CraftingGUITOGGLE")));
+        });

```

Comment:
What does this do exactly? Can we avoid creating those arraylists if we can just do a few comparisons? It would be *slightly* faster.

Reply:
I can try removing one...
The otherone is nessesary for parsing from ArrayList<String> to Doubele